lib/src/git.rs

```diff
@@ -159,13 +190,35 @@ pub fn import_some_refs(
         };
         let id = CommitId::from_bytes(git_commit.id().as_bytes());
         pinned_git_heads.insert(full_name.to_string(), vec![id.clone()]);
-        if !git_ref_filter(&full_name) {
-            continue;
-        }
+
+        let allow_resurrection = match git_ref_filter(&full_name) {
+            ImportMethod::Skip => continue,
+            ImportMethod::Merge => false,
+            ImportMethod::MergeOrResurrect => true,
+        };
         // TODO: Make it configurable which remotes are publishing and update public
         // heads here.
-        let old_target = jj_view_git_refs.remove(&full_name);
         let new_target = Some(RefTarget::Normal(id.clone()));
+        let old_target = jj_view_git_refs.remove(&full_name);
+        let old_target = if allow_resurrection
+            && new_target == old_target
```

Comment:
I'm considering removing the `&& new_target == old_target` condition here. That fixes the bug I described in https://github.com/martinvonz/jj/pull/1714#issuecomment-1601981672.

I'm not awake enough right this second to decide whether that can cause any problems, but it doesn't break any tests (other than the bug that's fixed).

Reply:
I think that this would be OK. Moreover, it seems to me at the moment that we can use `ImportMethod::MergeOrResurrect` whenever we import a remote-tracking branch, since it would be empty in only two cases: 1) if it was never fetched or 2) after `jj git forget`.

So, I'm thinking of rewriting this to no longer have  `ImportMethod::MergeOrResurrect`, but instead allowing resurrection of any remote-tracking branches.

Reply:
> I'm considering removing the `&& new_target == old_target` condition here.

SG. It probably behaves as if remote tracking branches in git repo were forgotten by `jj git forget`.

One possible problem (untested):
```shell
$ jj git fetch --remote foo
$ jj branch forget branch_originating_foo
$ jj git fetch --remote bar  # maybe resurrect "branch_originating_foo"?
```

I still think the underlying data model of `branch forget` is wonky, and would be better off redesigning it.

Reply:
OK, made this change and adjusted the comments.

> One possible problem (untested):

Excellent point, thanks! I think I fixed it by changing the `starts_with` condition in `fetch`. I don't feel like writing a test for that, it's a bit tedious.

> I still think the underlying data model of branch forget is wonky, and would be better off redesigning it.

:+1: I think it wouldn't actually be that hard. I may or may not get to it (I'm a bit tired of branches).